src/main/java/io/cdap/plugin/salesforce/plugin/source/batch/SalesforceBulkRecordReader.java

```diff
@@ -224,4 +257,17 @@ private String[] waitForBatchResults(BulkConnection bulkConnection)
     }
     throw new BulkAPIBatchException("Timeout waiting for batch results", info);
   }
+
+  private String[] getQueryResultList(BulkConnection bulkConnection) throws SalesforceQueryExecutionException {
+    try {
+      return bulkConnection.getQueryResultList(jobId, batchId).getResult();
+    } catch (AsyncApiException exception) {
+      LOG.error("The bulk query job {} failed.", jobId);
+      if (RETRY_ON_REASON.contains(exception.getExceptionCode())) {
+        throw new SalesforceQueryExecutionException(exception.getMessage());
+      }
```

Comment:
error logged

Reply:
This way you are loosing all the additional information from original AsyncApiException stack trace. There are only some case where you would want to hide all details of the underlying exception by using `e.getMessage` instead of attaching a root cause. I don't see why would we want to have it here.

Same applies to all the places in this PR.